service/service.go

```diff
@@ -158,6 +158,51 @@ func (s *Service) CreateArtifactType(ctx context.Context, req *api.CreateArtifac
 	}, nil
 }
 
+// UpdateArtifactType update specified artifact type.
+// Currently, this function is used to add new fields to artifact type.
+// Deleting field is not supported and all fields must be matched.
+func (s *Service) UpdateArtifactType(ctx context.Context, req *api.UpdateArtifactTypeRequest) (*api.UpdateArtifactTypeResponse, error) {
+	var putTypeOptions mlmetadata.PutTypeOptions
+
+	if req.ArtifactType == nil {
+		return nil, errors.New("no ArtifactType specified")
+	}
+
+	if req.PutTypeOptions == nil {
+		return nil, errors.New("no PutTypeOptions specified")
+	}
+
+	if req.PutTypeOptions.CanDeleteFields {
```

Comment:
The only acceptable PutTypeOptions now is

```
putTypeOptions {
					CanAddFields:       true
					CanDeleteFields:    false
					AllFieldsMustMatch: true}
```

How about not exposing PutTypeOptions in the API and add it back later if more possible configs are supported?

Reply:
I'm OK with it. We shall expose this later.